src/test/kotlin/com/_404/cosmos/springpicnic/ScenarioTests.kt

```diff
@@ -0,0 +1,71 @@
+package com._404.cosmos.springpicnic
+
+import com._404.cosmos.springpicnic.scenario.entity.Scenario
+import com._404.cosmos.springpicnic.scenario.repository.ScenarioRepository
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+
+@Suppress("No" +
+        "nAsciiCharacters", "TestFunctionName")
+@DataJpaTest
+class ScenarioTests {
+
+    @Autowired
+    lateinit var scenarioRepository: ScenarioRepository
+
+    val scenario1: Scenario = Scenario(id=1, gameSpan = 10, loadSize = 20)
+    val scenario2: Scenario = Scenario(id=2, gameSpan = 15, loadSize = 30)
+
+    @BeforeEach
+    fun setUp(){
+        scenarioRepository.deleteAll()
+    }
+
+    @Test
+    fun 시나리오를_추가_한다(){
+        scenarioRepository.save(scenario1)
+
+        Assertions.assertEquals(1, scenarioRepository.count())
+    }
+
+    @Test
+    fun 존재하지_않는_시나리오를_조회_한다(){
+        val scenario = scenarioRepository.findById(-1)
+        Assertions.assertFalse(scenario.isPresent)
+    }
+
+    @Test
+    fun 시나리오를_추가_및_조회_한다(){
+        scenarioRepository.save(scenario1)
+        scenarioRepository.save(scenario2)
+
+        Assertions.assertEquals(2, scenarioRepository.count())
+
+        var scenario = scenarioRepository.findById(scenario1.id!!)
+        Assertions.assertTrue(scenario.isPresent)
+
+        isSameScenario(scenario1, scenario.get())
+
+        scenario = scenarioRepository.findById(scenario2.id!!)
+        Assertions.assertTrue(scenario.isPresent)
+
+        isSameScenario(scenario2, scenario.get())
+
+    }
+
+    @Test
+    fun 모든_데이터를_지운다(){
+        scenarioRepository.deleteAll()
+        Assertions.assertEquals(0, scenarioRepository.count())
+    }
+
+    fun isSameScenario( scenario_a: Scenario,  scenario_b: Scenario){
+        Assertions.assertEquals(scenario_a.id, scenario_b.id)
+        Assertions.assertEquals(scenario_a.gameSpan, scenario_b.gameSpan)
+        Assertions.assertEquals(scenario_a.loadSize, scenario_b.loadSize)
+    }
```

Comment:
Domain Driven Development 관점에서
Scenario 클래스쪽에 equals 함수를 추가해서 쓰는것도 좋을거같음~
```suggestion
    fun isSameScenario( scenario_a: Scenario,  scenario_b: Scenario){
        Assertions.assertTrue(scenario_a == scenario_b)
    }
```

Reply:
코틀린에선 널체크를 어떻게 하는게 좋을지 고민해봤는데..
`id?.let { it == other.id } ?: false`
이런식으로 생각해봤다가
그냥 Scenario 클래스에 아래처럼 추가하면 될거같음 ㅎ

https://kotlinlang.org/docs/scope-functions.html#run
요건 내가 let, also, run .. 헷갈려서 이번 기회에 찾아본 링크 참고용으로 공유함!
```
override fun equals(other: Any?): Boolean {
	if (other !is Scenario) {
		return false
	}
	return this.id == other.id
			&& this.gameSpan == other.gameSpan
			&& this.loadSize == other.loadSize
}
```